### PR TITLE
search notebook: homepage onboarding experiment

### PR DESCRIPTION
**Pull request by novoselrok**
Preview:
Screenshot 2021-08-09 at 07 23 27

Issue #23096

### CODE REVIEW

**Review by rrhyne — APPROVED** [review]
LGTM!

File: client/web/src/featureFlags/featureFlags.ts [hunk 1]
```diff
@@ -6,9 +6,7 @@ import { dataOrThrowErrors, gql } from '@sourcegraph/shared/src/graphql/graphql'
 import { requestGraphQL } from '../backend/graphql'
 import { FetchFeatureFlagsResult } from '../graphql-operations'
 
-// This should be a `|` separated union of feature flag strings,
-// but right now we don't have any feature flags being tested.
-export type FeatureFlagName = never
+export type FeatureFlagName = 'search-notebook-onboarding'
```
Comment (limitedmage) [inline client/web/src/featureFlags/featureFlags.ts:9]: Suggested comment that does not have to change if we delete all feature flags:

```suggestion
// A union of strings of all feature flags we currently have.
// If there are no feature flags at the moment, this should be `never`.
export type FeatureFlagName = 'search-notebook-onboarding'
```

File: client/web/src/search/input/LoggedOutHomepage.tsx [hunk 2]
```diff
@@ -35,14 +36,88 @@ const exampleQueries = [
     },
 ]
 
-export interface LoggedOutHomepageProps extends TelemetryProps, ThemeProps {}
+const exampleNotebooks = [
```
Comment (limitedmage) [inline client/web/src/search/input/LoggedOutHomepage.tsx:39]: nit: add typing for clarity? (and same for `exampleQueries` above?)

```suggestion
const exampleNotebooks: SearchExample[] = [
```

**Review by limitedmage — APPROVED** [review]
Looks great!

File: client/web/src/search/input/LoggedOutHomepage.tsx [hunk 3]
```diff
@@ -137,6 +196,15 @@ export const LoggedOutHomepage: React.FunctionComponent<LoggedOutHomepageProps>
     )
 }
 
+const NotebookIcon = React.memo(() => (
```
Comment (limitedmage) [inline client/web/src/search/input/LoggedOutHomepage.tsx:199]: This icon looks more like a calendar event than a notebook. I'm wondering if something like MDI's `notebook` or `open-book-variant` icons might be a better choice? CC @rrhyne

File: client/web/src/search/input/SearchPage.story.tsx [hunk 4]
```diff
@@ -63,6 +63,7 @@ const defaultProps = (props: ThemeProps): SearchPageProps => ({
     hasUserAddedRepositories: false,
     hasUserAddedExternalServices: false,
     getUserSearchContextNamespaces: mockGetUserSearchContextNamespaces,
+    featureFlags: new Map(),
```
Comment (limitedmage) [inline client/web/src/search/input/SearchPage.story.tsx:66]: Can we add a story for the unauthenticated homepage with this new feature flag on?

File: client/web/src/search/input/LoggedOutHomepage.tsx (hunk 3 above)
Comment (novoselrok) [reply client/web/src/search/input/LoggedOutHomepage.tsx:199]: I was thinking about this as well. Here are some potential replacements:

Screenshot 2021-08-10 at 07 34 12
Screenshot 2021-08-10 at 07 35 18

Comment (rrhyne) [reply client/web/src/search/input/LoggedOutHomepage.tsx:199]: I'd use to book-outline: image

or notebook-outline:
image

Personally, notebook-outline is super busy to me. 

Perhaps we can customize it later with a magnifying glass in the lower right-hand corner.

**Comment by novoselrok** [comment]
With updated icon:

Screenshot 2021-08-10 at 16 04 29

**Pull request merged**